### PR TITLE
Document support for Python versions in pyproject.toml

### PR DESCRIPTION
**Pull request by jdufresne**

### CODE REVIEW

**Review by peternewman — CHANGES_REQUESTED** [review]
Can you rebase please. I think the GitHub workflow bit has been done, setup.py is now pyproject.toml, but the programming language tags aren't in that, so may still be relevant to add.

File: pyproject.toml [hunk 1]
```diff
@@ -17,7 +17,14 @@ classifiers = [
...
     "Operating System :: Unix",
-    "Operating System :: MacOS"
+    "Operating System :: MacOS",
+    "Programming Language :: Python",
+    "Programming Language :: Python :: 3",
+    "Programming Language :: Python :: 3 :: Only",
+    "Programming Language :: Python :: 3.7",
+    "Programming Language :: Python :: 3.8",
+    "Programming Language :: Python :: 3.9",
+    "Programming Language :: Python :: 3.10",
```
[the remark this answers is not in the document]
Comment (jdufresne) [reply pyproject.toml:27]: Just a personal anecdote, I have not found maintaining this list a hardship in other projects. When adding a new version to GitHub actions, this is just one more spot. Often, volunteers see this information and add it (such as myself).

The classifiers are useful as they display on PyPI to provide quick at-a-glance info to people interested in the project as to what Python versions are supported. They could also be used by scripts to determine trends in the Python community

Is there an existing wiki article I could edit to help?

[the remark this answers is not in the document]
Comment (peternewman) [reply pyproject.toml:27]: > Just a personal anecdote, I have not found maintaining this list a hardship in other projects. When adding a new version to GitHub actions, this is just one more spot. Often, volunteers see this information and add it (such as myself).

Yeah, I guess it more seems a shame we have to generate/manage them at all when we've already got a computer readable minimum version field!

> The classifiers are useful as they display on PyPI to provide quick at-a-glance info to people interested in the project as to what Python versions are supported. They could also be used by scripts to determine trends in the Python community

Again it seems slightly mad the other fields can't just be used. :shrug: 

> Is there an existing wiki article I could edit to help?

Not currently, I'd say just create a new one about adding or removing support for a new version (maybe even mention to check that before cutting a new release). There's clearly already two places in this file it needs updating, plus our various CI and any readme docs...

[the remark this answers is not in the document]
Comment (jdufresne) [reply pyproject.toml:27]: @peternewman I added the information to the following wiki article. Pleaese let me know if you'd like to see any changes to the wording.

https://github.com/codespell-project/codespell/wiki/How-to-make-a-release

**Comment by jdufresne** [comment]
Updated to include Python 3.11.

**Review by DimitriPapadopoulos — APPROVED** [review]

File: pyproject.toml (hunk 1 above)
[the remark this answers is not in the document]
Comment (peternewman) [reply pyproject.toml:27]: Great thanks, I've added a few further nudges.

**Review by peternewman — APPROVED** [review]
I think I'm mostly convinced the admin overheads aren't too arduous.

Over to @larsoner or @DimitriPapadopoulos if its actually widely used/worth adding...

**Comment by larsoner** [comment]
Thanks @jdufresne !

**Pull request merged**